### PR TITLE
refactor: Enable noImplicitAny

### PR DESCRIPTION
**Pull request by BANG88**
See #5627

### CODE REVIEW

File: components/anchor/index.tsx [hunk 1]
```diff
@@ -85,26 +85,26 @@ export default class Anchor extends React.Component<AnchorProps, any> {
...
-    if (child.type.__ANT_ANCHOR_LINK && href) {
-      this.anchorHelper.addLink(href);
-      return React.cloneElement(child, {
-        onClick: this.clickAnchorLink,
-        prefixCls: this.props.prefixCls,
-        bounds: this.props.bounds,
-        affix: this.props.affix || this.props.showInkInFixed,
-        offsetTop: this.props.offsetTop,
-      });
+  renderAnchorLink = (child: React.ReactElement<any>) => {
```
Comment (bang88) [inline components/anchor/index.tsx]: I found a same name function in anchorlink.tsx, do not know what's the difference between them ?

**Comment by benjycui** [comment]
Actually, I recommend you not to modify too many components in one PR. So that we can review your PR easier. Thanks.

**Comment by bang88** [comment]
I do agree. Just PR these components here we can talk about it, there are too many works need to do.

**Comment by benjycui** [comment]
👌 Just remove `[WIP]` when you think it's OK.

**Comment by bang88** [comment]
Get. don't know why the tests not pass.

**Review by benjycui — APPROVED** [review]
👍

**Review by infeng — APPROVED** [review]
:thumbsup:

**Pull request merged**